src/test/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSparkUnitTest.java

```diff
@@ -0,0 +1,314 @@
+        final File expectedCorrectedFile = new File(TEST_DATA_DIR, filenamePrefix + ".pp.ec.fa");
+        final String correctedFileName = compareNamesAndComputeSeqEditDist(actualCorrectedFile, expectedCorrectedFile, true, editDistancesBetweenSeq);
+        for(final Integer d : editDistancesBetweenSeq){ Assert.assertEquals(d, zero); }
+
+        editDistancesBetweenSeq = new ArrayList<>();
+        final File correctedFile = new File(workingDir, correctedFileName);
+        final File actualFilterPassingFile = new File(workingDir, RunSGAViaProcessBuilderOnSpark.runSGAFilter(sgaPath, correctedFile, workingDir, threads, runtimeInfo));
+        final File expectedFilterPassingFile = new File(TEST_DATA_DIR, filenamePrefix + ".pp.ec.filter.pass.fa");
+        final String filterPassingFileName = compareNamesAndComputeSeqEditDist(actualFilterPassingFile, expectedFilterPassingFile, true, editDistancesBetweenSeq);
+        for(final Integer d : editDistancesBetweenSeq){ Assert.assertEquals(d, zero); }
+
+        editDistancesBetweenSeq = new ArrayList<>();
+        final File filterPassingFile = new File(workingDir, filterPassingFileName);
+        final File actualRmdupFile = new File(workingDir, RunSGAViaProcessBuilderOnSpark.runSGARmDuplicate(sgaPath, filterPassingFile, workingDir, threads, indexer, indexerArgs, runtimeInfo));
+        final File expectedRmdupFile = new File(TEST_DATA_DIR, filenamePrefix + ".pp.ec.filter.pass.rmdup.fa");
+        final String duplicateRemovedFileName = compareNamesAndComputeSeqEditDist(actualRmdupFile, expectedRmdupFile, false, editDistancesBetweenSeq);
+        for(final Integer d : editDistancesBetweenSeq){ Assert.assertEquals(d, zero); }
+
+        editDistancesBetweenSeq = new ArrayList<>();
+        final File duplicateRemovedFile = new File(workingDir, duplicateRemovedFileName);
+        final File actualMergedFile = new File(workingDir, RunSGAViaProcessBuilderOnSpark.runSGAFMMerge(sgaPath, duplicateRemovedFile, workingDir, threads, indexer, indexerArgs, runtimeInfo));
+        final File expectedMergedFile = new File(TEST_DATA_DIR, filenamePrefix + ".pp.ec.filter.pass.rmdup.merged.fa");
+        final String mergedFileName = compareNamesAndComputeSeqEditDist(actualMergedFile, expectedMergedFile, false, editDistancesBetweenSeq);
+        for(final Integer d : editDistancesBetweenSeq){ Assert.assertEquals(d, zero); }
+
+        // final assembled contig test
+        final File mergedFile = new File(workingDir, mergedFileName);
+        final File actualAssembledContigsFile = new File(workingDir, RunSGAViaProcessBuilderOnSpark.runSGAOverlapAndAssemble(sgaPath, mergedFile, workingDir, threads, runtimeInfo));
+
+        final List<String> actualFASTAContents = (null==actualAssembledContigsFile) ? null : Files.readAllLines(Paths.get(actualAssembledContigsFile.getAbsolutePath()));
```

Comment:
\[FindBugs\] INFO: RCN: Redundant nullcheck of actualAssembledContigsFile, which is known to be non\-null in org\.broadinstitute\.hellbender\.tools\.spark\.sv\.RunSGAViaProcessBuilderOnSparkUnitTest\.stepByStepTestWorker\(File, File, File\)